glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/GleanLifecycleObserver.kt

```diff
@@ -4,20 +4,20 @@
 
 package mozilla.telemetry.glean.scheduler
 
-import android.content.Context
 import androidx.lifecycle.Lifecycle
 import androidx.lifecycle.LifecycleEventObserver
 import androidx.lifecycle.LifecycleOwner
+import mozilla.telemetry.glean.Dispatchers
 import mozilla.telemetry.glean.Glean
 import mozilla.telemetry.glean.GleanMetrics.GleanBaseline
+import mozilla.telemetry.glean.rust.LibGleanFFI
+import mozilla.telemetry.glean.rust.toByte
 
 /**
  * Connects process lifecycle events from Android to Glean's handleEvent
  * functionality (where the actual work of sending pings is done).
  */
-internal class GleanLifecycleObserver(
-    private val applicationContext: Context
```

Comment:
What was the reason we took in the context?

Reply:
We used to need the `Context` because we were using `SharedPreferences`, [in this PR](https://github.com/mozilla/glean/commit/1338a1dbbd3021e17b35bc848859c4daff6e0803#diff-66b8e9961a42e4203fd9a68980025f35R21), to save the state of the dirty bit. We are now doing this in Rust :-)